nixos/lib/utils.nix

```diff
@@ -21,8 +21,8 @@ rec {
   toShellPath = shell:
     if types.shellPackage.check shell then
       "/run/current-system/sw${shell.shellPath}"
-    else if types.package.check shell then
-      throw "${shell} is not a shell package"
+    else if isString shell && types.path.check shell then
```

Comment:
`path` accepts not only strings, but also derivations. Here we have already filtered derivations with `shellPath` defined (`shellPackage.check shell == true`), so now we want to handle strings that are also valid paths (start with `/`), but _not_ derivations. We'll then throw on them.